### DISCUSSION TITLE
x-for question

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [alpinejs/alpine#1311] [1. Help]
Below is my code:

`<ul x-data="{items:[{id:0,name: '512 Flex black jean
      Tapered slim fit',price:98,comp:'1% ELASTANE, 31% LYOCELL, 68% COTTON',url:'https://imagescdn.simons.ca/images/8543/21110/1/A1_1.jpg?__=2'},
      {id:1,name:'Faded T-shirt',price:35,comp:'COTTON',url:'https://imagescdn.simons.ca/images/6304/4918/10/A1_1.jpg?__=1'}]}"
        class="grid grid-col-2 gap-16"
      >
        <template x-for="item in items" :key="item.id">
          <li class="shadow-md" x-text="item.name"></li>
        </template>
      </ul>`

but when i open the page, I didn'see any list items
 sorry if the code looks messy

**Comment by sqllyw** [comment]
it works:
https://jsfiddle.net/8Luzojt4/

**Reply by Chandler-Zhu** [reply]
but when I put an image in the item like this, the image doesn't show up

`            <img
              class="h-96 w-full object-cover"
              src="item.url"
            />`

**Reply by ryangjchandler** [reply]
@Chandler-Zhu You need to use `x-bind:src="item.url"`.

**Reply by Chandler-Zhu** [reply]
I tried but this time I got a blank page
https://jsfiddle.net/zoLk9p8d/1/

**Reply by ryangjchandler** [reply]
You have an error in your syntax according to the browser console.

**Reply by sqllyw** [reply]
remove the char between 'jean' and 'Tapered' ,  it seems working:
https://jsfiddle.net/04rnxujo/

**Reply by Chandler-Zhu** [reply]
does this counts as a bug ？

**Reply by ryangjchandler** [reply]
It's likely an issue in your component, not Alpine.

**Reply by sqllyw** [reply]
> does this counts as a bug ？

I don't think its related to Alpine, you have to use 'jean \n Tapered' if you want new line, you can't directly embed a new line character inside a string, new line is to indicate the end of a statement, i might be wrong.

this non alphine code works:

https://jsfiddle.net/rmvoqyj9/

this code following your example string fails:

https://jsfiddle.net/rmvoqyj9/1/
